src/lib.rs

```diff
@@ -368,49 +271,50 @@ pub struct Request {
     /// Total number of bytes from the start of the HTTP request message until (and including)
     /// the double CRLF before the body.
     /// Set to -1 if the info is not available.
-    headers_size: Option<int>,
+    headers_size: Option<i32>,
 
     /// Size of the request body (POST data payload) in bytes.
     /// Set to -1 if the info is not available.
-    body_size: Option<int>,
+    body_size: Option<i32>,
 
     /// A comment provided by the user or the application.
     comment: Option<String>
 }
 
-impl <S: Encoder<E>, E> Encodable<S, E> for Request {
-    fn encode(&self, encoder: &mut S) -> Result<(), E> {
-        encoder.emit_struct("Request", 0, |encoder| {
-            let default_int = -1i;
-            let mut fields: Vec<(&str, Box<Encodable<S, E>>)> = Vec::new();
-            fields.push(("method", box self.method.to_string()));
-            fields.push(("url", box self.url.to_string()));
-            fields.push(("httpVersion", box self.http_version.to_string()));
-            fields.push(("cookies", box self.cookies.as_slice()));
-            fields.push(("headers", box self.headers.as_slice()));
-            fields.push(("queryString", box self.query_string.as_slice()));
-            match self.post_data {
-                Some(ref post_data) => fields.push(("postData", box post_data)),
-                None => ()
-            }
-            fields.push(("headersSize", box self.headers_size.unwrap_or(default_int)));
-            fields.push(("bodySize", box self.body_size.unwrap_or(default_int)));
-            match self.comment {
-                Some(ref comment) => fields.push(("comment", box comment.to_string())),
-                None => ()
-            }
-            for (idx, &(name, ref value)) in fields.iter().enumerate() {
-                try!(encoder.emit_struct_field(name, idx, |encoder| (**value).encode(encoder)));
-            }
-            Ok(())
-        })
+impl Request {
+    pub fn new(
+        method: String,
+        url: String,
+        http_version: String,
+        cookies: Vec<Cookie>,
+        headers: Vec<Header>,
+        query_string: Vec<QueryStringPair>,
+        post_data: Option<PostData>,
+        headers_size: Option<i32>,
+        body_size: Option<i32>,
+        comment: Option<String>
+    ) -> Request {
+        Request {
+            method: method,
+            url: url,
+            http_version: http_version,
+            cookies: cookies,
+            headers: headers,
+            query_string: query_string,
+            post_data: post_data,
+            headers_size: headers_size,
+            body_size: body_size,
+            comment: comment
+        }
     }
 }
 
 /// This object contains detailed info about the response.
+#[derive(Serialize, Deserialize, PartialEq, Debug)]
+#[serde(rename_all = "camelCase")]
```

Comment:
The [specification](http://www.softwareishard.com/blog/har-12-spec/#response) says that the redirect_url field should be redirectURL in the final JSON, but this turns it into redirectUrl.